src/lib.rs

```diff
@@ -0,0 +1,350 @@
+//! A Bech32m address encoder/decoder for PQC public keys.
+//!
+//! This crate supports:
+//! - Mainnet vs Testnet via HRP.
+//! - Disjoint byte ranges for version and pubkey type.
+//! - Support for multiple PQC public keys.
+//! - Full encode/decode round‑trip with Bech32m checksum per BIP‑350.
+
+use bech32::{Bech32m, Hrp, decode, encode};
+use hex::encode as hex_encode;
+use sha2::{Digest as ShaDigest, Sha256};
+use std::convert::TryFrom;
+use thiserror::Error;
+
+/// A Bech32 string is at most 90 characters long [BIP-173]
+/// PQ address length is 64 characters.
+const ADDRESS_LENGTH: usize = 64;
```

Comment:
This says 64, but then a couple of the error types say 67, e.g.
```
/// A PQ address is 67 characters long
    #[error("A PQ address is 67 characters long: got {0}")]
    InvalidEncodingLength(usize),
```
Which of them is correct?